### ISSUE TITLE
No clientID, clientSecret fields in Settings section

### ISSUE THREAD

**Issue by Rob-ee** [tonesto7/nest-manager#553]
## Verify the following before opening an issue:

- [X] You followed the Nest Developer documentation here? [Nest Developer Account Creation](https://cdn.rawgit.com/tonesto7/nest-manager/master/Documents/html/Create_Dev_Account.html)
- [X] That _OAuth_ is Enabled for the SmartApp under the IDE.
- [X] The SmartApps and *_ALL_* (5) Devices Handlers are using the latest code available.
- [X] You have reviewed the [NST Manager Wiki](http://thingsthataresmart.wiki/index.php?title=NST_Manager) for potential fixes.
- [X] That ALL SmartApps and Device Handlers have been _Published for You_ in the IDE.

## About Your Setup:
* How many Thermostats?: 1
* How many Protects? 0
* How many Cameras? 0
* Using the Nest Presence Device? 1
* Using the Nest Weather Device? 1
* iOS or Android?: Android
* Mobile App Version:

## Expected Behavior
In ST IDE, for Nest Manager SmartApp, according to PDF , there should be clientID & ClientSecret fields under "Settings" section

## Current Behavior
There are no fields under "Settings" section, just text "Add setting definitions in the source code and then set the values here"

## Steps to Reproduce (for bugs)
1. Installed Nest manager via auto ST Community installer
2. Followed instructions in https://rawgit.com/tonesto7/nest-manager/master/Documents/instructions/Nest%20Developer%20-%20Nest%20Product.pdf and created NEST dev account, copied Product ID, Product Secret to clipboard
3. Opened the "Settings" section of Nest Manager SmartApp, but there are no fields, just a message "Add setting definitions in the source code and then set the values here"
4.

**Comment by Watso4183** [comment]
Ran into the same issue myself. Installed via ST Community installer as well.

**Comment by MarkToon** [comment]
Did this ever get solved? Ive ran into the same issue myself.

**Comment by safetymonkey** [comment]
+1. I'm having the same issue.

**Comment by Rob-ee** [comment]
I was playing around with it and honestly I can't remember exactly what 
I did to resolve it, but I think it had something to do with manually 
downloading each script and running it, I think that I figured out that 
the bundled installer was a bit different than the last version

**Comment by safetymonkey** [comment]
Well, sure. The problem doesn't exist if I download the raw versions of the
scripts and input them in the IDE directly. But this bug relates
specifically to the app installer, which is listed as the "strongly
recommended" option. :)

**Comment by ramginuga** [comment]
+1, same issue

**Comment by cdevereu** [comment]
+1 As a new ST user this is my first community installed smart app, following the "Strongly recommended" option.

**Comment by bbuchan10** [comment]
I resolved this by clicking "code" in the right corner in the IDE then Clicking save. That seemed to have triggered the IDE to re-digest the definitions.

**Comment by cdevereu** [comment]
I also resolved this by manually copying/pasting the Nest Manager raw code from Github (https://github.com/tonesto7/nest-manager/blob/master/smartapps/tonesto7/nest-manager.src/nest-manager.groovy) over the top of the existing code put there by the community installer.  Save > Publish then the settings appeared

**Comment by safetymonkey** [comment]
Again, copying the code defeats the purpose of the recommended installation
method. I can and have copied the code over, but this is a bug with the
recommended solution.

**Comment by bbuchan10** [comment]
I do not believe it is an issue with the automated installation. As the definitions for the fields are in the code. Just simply saving the code again, not pasting over it(that is extra work), resolves the issue. I believe it is an issue with the IDE ingesting the definition the first time.

**Comment by safetymonkey** [comment]
Then the instructions should probably be updated to note this workaround.

**Comment by bugzc** [comment]
Same issue here. I can confirm going into code and hitting 'save' did the trick. A note in the Wiki would be helpful.

**Comment by automate2win** [comment]
Had the same issue as everyone else above and going to the code and hitting "save" worked for me as well.  I agree with bugzc

**Comment by VTChevalier** [comment]
Yes - agree the guide for creating your own nest dev id (https://rawgit.com/tonesto7/nest-manager/master/Documents/instructions/Nest%20Developer%20-%20Nest%20Product.pdf) needs to be updated to mention quickly saving the code in IDE to allow the settings to refresh, wasted a few mins on this myself.

**Comment by dcurrie1961** [comment]
@cdevereu Your solution worked for me. I did a faked an upgrade and republish. Settings appeared and it was fine.
